### PR TITLE
Add kotlin.math for wasm

### PR DESCRIPTION
**Pull request by ilmat192**

### CODE REVIEW

File: Interop/JsRuntime/build.gradle [hunk 1]
```diff
@@ -1,39 +1,20 @@
-import org.jetbrains.kotlin.KlibInstall
-
-apply plugin: 'konan'
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
```
Comment (alexander-gorshenev) [inline Interop/JsRuntime/build.gradle]: 2018

Comment (ilmat192) [reply Interop/JsRuntime/build.gradle]: done

File: backend.native/tests/runtime/math/wasm.kt [hunk 2]
```diff
@@ -0,0 +1,1623 @@
+/*
```
[the remark this answers is not in the document]
Comment (ilmat192) [reply backend.native/tests/runtime/math/wasm.kt]: done

File: runtime/src/main/js/math.js [hunk 3]
```diff
@@ -0,0 +1,267 @@
+/*
```
[the remark this answers is not in the document]
Comment (ilmat192) [reply runtime/src/main/js/math.js]: done

File: Interop/JsRuntime/src/main/kotlin/jsinterop.kt [hunk 4]
```diff
@@ -26,9 +42,13 @@ external public fun pushIntToArena(arena: Arena, value: Int)
 
 const val upperWord = 0xffffffff.toLong() shl 32
 
+@Used
+@ExportForCppRuntime
```
Comment (olonho) [inline Interop/JsRuntime/src/main/kotlin/jsinterop.kt]: Shalln't @ExportForCppRuntime imply @Used?

File: runtime/build.gradle [hunk 5]
```diff
@@ -51,3 +51,29 @@ task clean {
...
+        def jsinterop = "$distDir/bin/$jsinteropScript"
+        def targetDir = "$buildDir/generated"
+        def firstLine = true
+        "$jsinterop -pkg kotlinx.interop.wasm.math -o $targetDir/math -target wasm32".execute().waitFor()
+        copy {
+            from("$targetDir/math-build/natives/js_stubs.js")
+            into('src/main/js')
+            rename { "math.js" }
+            filter { line ->
+                if (firstLine) {
```
Comment (olonho) [inline runtime/build.gradle]: Why so hacky?

File: runtime/src/main/cpp/Types.h [hunk 6]
```diff
@@ -50,6 +50,12 @@ typedef ObjHeader* KRef;
 typedef const ObjHeader* KConstRef;
 typedef const ArrayHeader* KString;
 
+namespace JS {
```
Comment (olonho) [inline runtime/src/main/cpp/Types.h]: Don't think it shall be done this way.

Comment (ilmat192) [reply runtime/src/main/cpp/Types.h]: done

File: runtime/build.gradle (hunk 5 above)
Comment (ilmat192) [reply runtime/build.gradle]: done

File: Interop/JsRuntime/src/main/kotlin/jsinterop.kt (hunk 4 above)
Comment (ilmat192) [reply Interop/JsRuntime/src/main/kotlin/jsinterop.kt]: No but these functions are not called from JS so we still can remove `@Used` here.

File: runtime/src/main/cpp/KotlinMath.h [hunk 7]
```diff
@@ -23,44 +23,42 @@
 
 extern "C" {
```
Comment (alexander-gorshenev) [inline runtime/src/main/cpp/KotlinMath.h]: // TODO: consider auto-generating this header file.

**Pull request merged**